billy-gin/src/main/java/com/premiumminds/billy/gin/services/impl/pdf/FOPPDFTransformer.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (C) 2017 Premium Minds.
+ *
+ * This file is part of billy GIN.
+ *
+ * billy GIN is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option) any
+ * later version.
+ *
+ * billy GIN is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with billy GIN. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.premiumminds.billy.gin.services.impl.pdf;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.StringReader;
+import java.net.URI;
+
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.sax.SAXResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.apache.fop.apps.FOPException;
+import org.apache.fop.apps.FOUserAgent;
+import org.apache.fop.apps.Fop;
+import org.apache.fop.apps.FopFactory;
+import org.apache.xmlgraphics.util.MimeConstants;
+
+import com.premiumminds.billy.gin.services.exceptions.ExportServiceException;
+import com.premiumminds.billy.gin.services.export.ParamsTree;
+import com.premiumminds.billy.gin.services.export.ParamsTree.Node;
+
+public abstract class FOPPDFTransformer {
+
+	private Source mapParamsToSource(ParamsTree<String, String> documentParams) {
+		StreamSource source = new StreamSource(new StringReader(
+				FOPPDFTransformer.generateXML(documentParams)));
+
+		return source;
+	}
+
+	public static String generateXML(ParamsTree<String, String> tree) {
+		String rval = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n";
+		return rval + FOPPDFTransformer.toXML(tree.getRoot());
+	}
+
+	private static String toXML(Node<String, String> node) {
```

Comment:
We should use a StringBuilder instance instead of string concatenation. In the toXML, the unique StringBuilder instance should be the first argument and passed by the upper layer. I also think it should be (return type) void and be renamed into writeXML